runtime/runtime/src/metrics.rs

```diff
@@ -255,25 +317,75 @@ impl ApplyMetrics {
             self.update_accumulated(accumulated_gas, accumulated_compute);
     }
 
-    pub fn local_receipts_done(&mut self, accumulated_gas: u64, accumulated_compute: u64) {
+    pub fn local_receipts_done(
+        &mut self,
+        count: u64,
+        time: Duration,
+        accumulated_gas: u64,
+        accumulated_compute: u64,
+    ) {
         (self.local_receipts_gas, self.local_receipts_compute_usage) =
             self.update_accumulated(accumulated_gas, accumulated_compute);
+        self.local_receipts_processed_total += 1;
+        self.local_receipts_processing_seconds_total += time.as_secs_f64();
     }
 
-    pub fn delayed_receipts_done(&mut self, accumulated_gas: u64, accumulated_compute: u64) {
+    pub fn delayed_receipts_done(
+        &mut self,
+        count: u64,
+        time: Duration,
+        accumulated_gas: u64,
+        accumulated_compute: u64,
+    ) {
         (self.delayed_receipts_gas, self.delayed_receipts_compute_usage) =
             self.update_accumulated(accumulated_gas, accumulated_compute);
+        self.delayed_receipts_processed_total += count;
+        self.delayed_receipts_processing_seconds_total += time.as_secs_f64();
     }
 
-    pub fn incoming_receipts_done(&mut self, accumulated_gas: u64, accumulated_compute: u64) {
+    pub fn incoming_receipts_done(
+        &mut self,
+        count: u64,
+        time: Duration,
+        accumulated_gas: u64,
+        accumulated_compute: u64,
+    ) {
         (self.incoming_receipts_gas, self.incoming_receipts_compute_usage) =
             self.update_accumulated(accumulated_gas, accumulated_compute);
+        self.incoming_receipts_processed_total += count;
+        self.incoming_receipts_processing_seconds_total += time.as_secs_f64();
     }
 
     /// Report statistics
-    pub fn report(&self, shard_id: &str) {
+    pub fn report(&mut self, shard_id: &str) {
```

Comment:
Where possible please use ShardUID to make it future proof (most notably for resharding as multiple shards with same ids exist at the same time). Not a blocker though.

Reply:
In the context of metrics, it is better to have this consistent notation for aggregation. We can work on a migration plan to add ShardUID to all metrics.